vyper/contracts/libs/ZoneInteraction.vy

```diff
@@ -0,0 +1,161 @@
+# @version ^0.3.3
+
+
+interface ZoneInteraction:
+    def isValidOrder(
+        orderHash: bytes32,
+        caller: address,
+        offerer: address,
+        zoneHash: bytes32
+    ) -> bytes4: view
+
+    def isValidOrderIncludingExtraData(
+        orderHash: bytes32,
+        caller: address,
+        order: AdvancedOrder,
+        priorOrderHashes: DynArray[bytes32, 10],
+        criteriaResolvers: DynArray[CriteriaResolver, 10]
+    ) -> bytes4: view
+
+
+struct OfferItem:
+    itemType: uint256
+    token: address
+    identifierOrCriteria: uint256
+    startAmount: uint256
+    endAmount: uint256
+
+
+struct ConsiderationItem:
+    itemType: uint256
+    token: address
+    identifierOrCriteria: uint256
+    startAmount: uint256
+    endAmount: uint256
+    recipient: address
+
+
+struct OrderParameters:
+    offerer: address
+    zone: address
+    offer: DynArray[OfferItem, 10]
+    consideration: DynArray[ConsiderationItem, 10]
+    orderType: uint256
+    startTime: uint256
+    endTime: uint256
+    zoneHash: bytes32
+    salt: uint256
+    conduitKey: bytes32
+    totalOriginalConsiderationItems: uint256
+
+
+struct AdvancedOrder:
+    parameters: OrderParameters
+    numerator: uint120
+    denominator: uint120
+    signature: Bytes[1024]
+    extraData: Bytes[1024]
+
+
+struct CriteriaResolver:
+    orderIndex: uint256
+    side: uint256
+    index: uint256
+    identifier: uint256
+    criteriaProof: DynArray[bytes32, 10]
```

Comment:
Length 10 is insufficient for this one — say you have a 10k collection and you want to make an offer to buy any item that hasn't been compromised (where there are 10 compromised items): that means you have 9990 valid identifiers and a proof size of `Math.ceil(Math.log2(9990)) == 14`. I'd say a length of 15 would suffice for nearly all applications (though it's not ideal, length of 20 would be a more conservative choice)